### ISSUE TITLE
Warnig on LaTeX build at master tip (7e851510e)

### ISSUE THREAD

**Issue by jfbu** [sphinx-doc/sphinx#5946]
I get a strange warning at master tip (current 7e851510e) on `make clean latex` (blank lines added to visualize)

```console
processing sphinx.tex...contents intro usage/installation usage/quickstart usage/restructuredtext/index usage/restructuredtext/basics usage/restructuredtext/roles usage/restructuredtext/directives usage/restructuredtext/field-lists usage/restructuredtext/domains /Users/xxxxxx/sphinx/doc/usage/restructuredtext/index.rst: 

WARNING: toctree contains ref to nonexisting file 'usage/restructuredtext/domains'

usage/markdown usage/configuration usage/builders/index usage/extensions/index usage/extensions/autodoc usage/extensions/autosectionlabel usage/extensions/autos
[...]
```
Remarks:

- `make clean html` emits no warning,

- I do see a file `usage/restructuredtext/domains.rst` on checking out master branch...

**Comment by jfbu** [comment]
Bisecting to d6b7aad13, but I don't see what's wrong.

**Comment by jfbu** [comment]
Reverting the changes from https://github.com/sphinx-doc/sphinx/commit/d6b7aad1389a728956e4d052e2113e5669d93edd **only for documentation file** `doc/usage/restructuredtext/domains.rst` fixes the issue. I will now try to localize the exact change. But this is weird anyway.

**Comment by jfbu** [comment]
OK, the problem goes away by applying following patch:

```diff
diff --git a/doc/usage/restructuredtext/domains.rst b/doc/usage/restructuredtext/domains.rst
index 621713264..343bc9af0 100644
--- a/doc/usage/restructuredtext/domains.rst
+++ b/doc/usage/restructuredtext/domains.rst
@@ -815,9 +815,6 @@ The following directive can be used for this purpose.
 
    becomes
 
-   .. cpp:alias:: Data::a
-                  overload_example::C::f
-
    whereas::
 
        .. cpp:alias:: void overload_example::C::f(double d) const
@@ -825,9 +822,6 @@ The following directive can be used for this purpose.
 
    becomes
 
-    .. cpp:alias:: void overload_example::C::f(double d) const
-                   void overload_example::C::f(double d)
-
 
 Constrained Templates
 ~~~~~~~~~~~~~~~~~~~~~
```

i.e. remove usage in the doc of new ``:cpp:alias:`` directive.

@jakobandersen what do you think?

**Comment by jfbu** [comment]
refs: #5901

**Comment by jakobandersen** [comment]
I also get the warning when compiling into singlehtml.
The ``cpp:alias`` is using a transform to expand the content after everything has been parsed, but before references are resolved (https://github.com/sphinx-doc/sphinx/blob/master/sphinx/domains/cpp.py#L6792). I guess the priority should be a lower value? What is the proper priority to based it on?

**Comment by tk0miya** [comment]
Note: I got this stacktrace from 2.0 HEAD now.
```
Traceback (most recent call last):
  File "/Users/tkomiya/work/sphinx/sphinx/util/nodes.py", line 425, in inline_all_toctrees
    colorfunc, traversed)
  File "/Users/tkomiya/work/sphinx/sphinx/util/nodes.py", line 414, in inline_all_toctrees
    tree = cast(nodes.document, tree.deepcopy())
  File "/Users/tkomiya/work/sphinx/.tox/docs/lib/python3.7/site-packages/docutils/nodes.py", line 1008, in deepcopy
    copy.extend([child.deepcopy() for child in self.children])
  File "/Users/tkomiya/work/sphinx/.tox/docs/lib/python3.7/site-packages/docutils/nodes.py", line 1008, in <listcomp>
    copy.extend([child.deepcopy() for child in self.children])
  File "/Users/tkomiya/work/sphinx/.tox/docs/lib/python3.7/site-packages/docutils/nodes.py", line 1008, in deepcopy
    copy.extend([child.deepcopy() for child in self.children])
  File "/Users/tkomiya/work/sphinx/.tox/docs/lib/python3.7/site-packages/docutils/nodes.py", line 1008, in <listcomp>
    copy.extend([child.deepcopy() for child in self.children])
  File "/Users/tkomiya/work/sphinx/.tox/docs/lib/python3.7/site-packages/docutils/nodes.py", line 1008, in deepcopy
    copy.extend([child.deepcopy() for child in self.children])
  File "/Users/tkomiya/work/sphinx/.tox/docs/lib/python3.7/site-packages/docutils/nodes.py", line 1008, in <listcomp>
    copy.extend([child.deepcopy() for child in self.children])
  File "/Users/tkomiya/work/sphinx/.tox/docs/lib/python3.7/site-packages/docutils/nodes.py", line 1008, in deepcopy
    copy.extend([child.deepcopy() for child in self.children])
  File "/Users/tkomiya/work/sphinx/.tox/docs/lib/python3.7/site-packages/docutils/nodes.py", line 1008, in <listcomp>
    copy.extend([child.deepcopy() for child in self.children])
  File "/Users/tkomiya/work/sphinx/.tox/docs/lib/python3.7/site-packages/docutils/nodes.py", line 1008, in deepcopy
    copy.extend([child.deepcopy() for child in self.children])
  File "/Users/tkomiya/work/sphinx/.tox/docs/lib/python3.7/site-packages/docutils/nodes.py", line 1008, in <listcomp>
    copy.extend([child.deepcopy() for child in self.children])
  File "/Users/tkomiya/work/sphinx/.tox/docs/lib/python3.7/site-packages/docutils/nodes.py", line 1008, in deepcopy
    copy.extend([child.deepcopy() for child in self.children])
  File "/Users/tkomiya/work/sphinx/.tox/docs/lib/python3.7/site-packages/docutils/nodes.py", line 1008, in <listcomp>
    copy.extend([child.deepcopy() for child in self.children])
  File "/Users/tkomiya/work/sphinx/.tox/docs/lib/python3.7/site-packages/docutils/nodes.py", line 1008, in deepcopy
    copy.extend([child.deepcopy() for child in self.children])
  File "/Users/tkomiya/work/sphinx/.tox/docs/lib/python3.7/site-packages/docutils/nodes.py", line 1008, in <listcomp>
    copy.extend([child.deepcopy() for child in self.children])
  File "/Users/tkomiya/work/sphinx/.tox/docs/lib/python3.7/site-packages/docutils/nodes.py", line 1007, in deepcopy
    copy = self.copy()
  File "/Users/tkomiya/work/sphinx/sphinx/util/nodes.py", line 519, in _new_copy
    newnode = self.__class__(self.rawsource, **self.attributes)
TypeError: __init__() got an unexpected keyword argument 'ids'

During handling of the above exception, another exception occurred:
```

**Comment by tk0miya** [comment]
#6131 will resolve this issue. Please check it :-)

**Comment by tk0miya** [comment]
Fixed by #6141 :-)
